src/python/pants/backend/docker/lint/hadolint/rules.py

```diff
@@ -25,7 +25,7 @@
 class HadolintFieldSet(FieldSet):
     required_fields = (DockerImageSourceField,)
 
-    source: DockerImageSourceField
+    sources: DockerImageSourceField
```

Comment:
> Fix regression after going from multiple sources to single source.

Oh, oops. Yeah. That indeed is a regression. I'll fix it right now in a different way because we need to cherry-pick the fix to Pants 2.8.

> Also noting, that currently only docker_image targets supports linting. Should add support for linting the dockerfile target directly as well.

Interesting! We generally weren't envisioning running linters/formatters on generated code, but indeed it makes sense to me to do here!

Reply:
This both fixes the regression and unblocks you from running hadolint on the generated source files https://github.com/pantsbuild/pants/pull/13456